flock_agent/daemon/daemon.py

```diff
@@ -357,3 +361,96 @@ def log(self, request, response, time):
         # Change permissions of unix socket
         os.chmod(self.unix_socket_path, 0o660)
         os.chown(self.unix_socket_path, 0, self.gid)
+
+    async def autoupdate_loop(self):
+        # Autoupdate is only available for macOS right now; Linux uses package managers
+        if Platform.current() != Platform.MACOS:
+            return
+
+        update_check_delay = 43200  # 12 hours
```

Comment:
Do we want it to check immediately on startup? Or are we fine with it waiting 12 hours?

Reply:
It is checking immediately on startup right now. This is just the delay in seconds, but the sleep itself happens after the check (each time there's a `continue`, like line 402 and 437).